tests/NHSD.BuyingCatalogue.API.IntegrationTests/Steps/Documents/DocumentsApiSteps.cs

```diff
@@ -14,87 +14,89 @@ namespace NHSD.BuyingCatalogue.API.IntegrationTests.Steps.Documents
-            };
-            await SendModel(model, _context).ConfigureAwait(false);
+            await SendModel(model, _context, ScenarioContextMappingKey).ConfigureAwait(false);
         }
 
         [Given(@"the document api fails with solutionId ([\w-]*)")]
         public async Task GivenTheDocumentApiFailsWithSolutionId(string solutionId)
         {
-            MappingModel model = new MappingModel
-            {
-                Response = new ResponseModel { StatusCode = 500, Body = "Demo Error" }
-            };
+            var model = CreateMappingModel($"/api/v1/solutions/{solutionId}/documents", 500, "Demo Error");
+
+            await SendModel(model, _context, ScenarioContextMappingKey).ConfigureAwait(false);
+        }
+
+        [Given(@"The document api is (up|down)")]
+        public async Task GivenTheDocumentApiIs(string state)
+        {
+            var model = CreateMappingModel("/health/live", state == "up" ? 200 : 404);
+
+            await SendModel(model, _context, ScenarioContextMappingKey).ConfigureAwait(false);
+        }
 
-            model.Request = new RequestModel
+        private static MappingModel CreateMappingModel(string path, int responseStatusCode, string responseBody = null)
+        {
+            return new MappingModel
             {
-                Path = new PathModel
+                Response = new ResponseModel { StatusCode = responseStatusCode, Body = responseBody },
+                Request = new RequestModel
                 {
-                    Matchers = new[]
+                    Path = new PathModel
                     {
-                        new MatcherModel
-                        {
-                            Name = "WildcardMatcher",
-                            Pattern = $"/api/v1/solutions/{solutionId}/documents",
-                            IgnoreCase = true
-                        }
-                    }
-                },
-                Methods = new[] { "GET" }
+                        Matchers = new[]
+                            {
+                                new MatcherModel
+                                {
+                                    Name = "WildcardMatcher",
+                                    Pattern = path,
+                                    IgnoreCase = true
+                                }
+                            }
+                    },
+                    Methods = new[] { "GET" }
+                }
             };
+        }
 
-            await SendModel(model, _context).ConfigureAwait(false);
+        private static async Task SendModel(MappingModel model, ScenarioContext context, string mappingKey)
+        {
+            await AddMapping(model).ConfigureAwait(false);
+
+            if (!context.ContainsKey(mappingKey))
+                context[mappingKey] = new List<Guid>();
+
+            if (context[mappingKey] is List<Guid> guidList)
+                guidList.Add(model.Guid.Value);
```

Comment:
resharper warning here - need to check guid is set.

Reply:
👍